platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/TransactionPoolNexus.java

```diff
@@ -132,12 +136,12 @@ public TransactionPoolNexus(@NonNull final PlatformContext platformContext) {
 
     /**
      * Attempt to submit an application transaction. Similar to
-     * {@link #submitTransaction(ConsensusTransactionImpl, boolean)} but with extra safeguards.
+     * {@link #submitTransaction} but with extra safeguards.
      *
      * @param transaction the transaction to submit
      * @return true if the transaction passed all validity checks and was accepted by the consumer
      */
-    public synchronized boolean submitApplicationTransaction(@NonNull final SwirldTransaction transaction) {
```

Comment:
becuase of the size check, its better if we accept `Bytes` and `StateSignaturePayload` instead of `OneOf`